evap/student/templates/student_index.html

```diff
@@ -81,18 +81,44 @@
+                                                <ul class="dropdown-menu" role="menu">
+                                                    {% for grade_document in course.grade_documents.all %}
+                                                        <li><a href="{% url "grades:download_grades" grade_document.id %}">{{ grade_document.description }}</a></li>
+                                                    {% endfor %}
+                                                </ul>
+                                            </div>
+                                        {% elif course.is_graded and can_download_grades %}
+                                            <div data-toggle="tooltip" data-placement="left" class="disabled-tooltip" title="{% trans "No grades have been uploaded yet." %}"><button type="button" class="btn btn-sm btn-default dropdown-toggle" disabled="disabled">{% trans "Download grades" %} <span class="caret"></span></button></div>
+                                        {% endif %}
+                                        {% if course.state == 'published' %}
+                                            {% if course.can_publish_grades %}
+                                                <a href="{% url "results:course_detail" semester.id course.id %}" class="btn btn-sm btn-default">{% trans "Results" %}</a>
+                                            {% else %}
+                                                <div data-toggle="tooltip" data-placement="left" class="disabled-tooltip" title="{% trans "We did not receive enough votes to publish the results of this course" %}"><a class="btn btn-sm btn-default" disabled="disabled">{% trans "Results" %}</a></div>
+                                            {% endif %}
+                                        {% endif %}
+                                    {% else %}
+                                        {% if course.preliminary_grade_documents.count > 0 and can_download_grades %}
+                                            <div class="btn-group">
+                                                <button type="button" class="btn btn-sm btn-default dropdown-toggle" data-toggle="dropdown" aria-expanded="false" {{ disable_if_archived }}>{% trans "Download grades" %} <span class="caret"></span></button>
+                                                <ul class="dropdown-menu" role="menu">
+                                                    {% for grade_document in course.preliminary_grade_documents %}
+                                                        <li><a href="{% url "grades:download_grades" grade_document.id %}">{{ grade_document.description }}</a></li>
+                                                    {% endfor %}
+                                                </ul>
+                                            </div>
```

Comment:
these ~10 lines are exactly the same as the ones starting at line 86, except for preliminary vs final. please check the comment above and merge these parts if possible.